### PR TITLE
Improve tracing/logging on MQTT protocol errors

### PR DESCRIPTION
**Pull request by calohmn**
For example concerning a case where a MQTT message exceeding the max message size was received.

### CODE REVIEW

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java [hunk 1]
```diff
@@ -1824,8 +1846,19 @@ protected final void close(final String reason, final boolean sendDisconnectedEv
...
+                span = newSpan("close connection due to protocol error");
+                TracingHelper.logError(span, protocolLevelException);
+                reason = "protocol error: " + protocolLevelException.toString();
+            } else if (stopCalled()) {
+                span = newSpan("close device connection (server shutdown)");
+                reason = "server shutdown";
+            } else {
+                span = newSpan("CLOSE");
+                reason = null;
+            }
```
Comment (sophokles73) [inline adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java:1861]: FMPOV using if - else if - else makes the code so much more readable than when using the conditional assignment :+1:

File: tests/src/test/java/org/eclipse/hono/tests/mqtt/MqttPublishTestBase.java [hunk 2]
```diff
@@ -374,6 +375,51 @@ public void testUploadMessagesWithWrongContentTypeSendsErrors(final VertxTestCon
...
+        }
+
+        final byte[] payloadBytes = IntegrationTestSupport.getPayload(ServiceConfigProperties.DEFAULT_MAX_PAYLOAD_SIZE);
+        final Promise<Void> clientClosedPromise = Promise.promise();
+        // GIVEN a connected device
+        connectToAdapter(IntegrationTestSupport.getUsername(deviceId, tenantId), "secret")
+                .compose(connAck -> {
+                    mqttClient.closeHandler(v -> clientClosedPromise.complete());
+                    // WHEN a device publishes a message with a payload exceeding the max size
+                    return send(tenantId, deviceId, Buffer.buffer(payloadBytes), false, null);
```
Comment (sophokles73) [inline tests/src/test/java/org/eclipse/hono/tests/mqtt/MqttPublishTestBase.java:409]: shouldn't a payload of size ServiceConfigProperties.DEFAULT_MAX_PAYLOAD_SIZE actually be accepted?

Comment (calohmn) [reply tests/src/test/java/org/eclipse/hono/tests/mqtt/MqttPublishTestBase.java:409]: We are setting the `MqttServerOptions#maxMessageSize` on the MQTT server here. This size consists of the payload size plus the size of the variable header, containing the topic name (e.g. `command/DEFAULT_TENANT/4711/res/1010f8ab0b53-bd96-4d99-9d9c-56b868474a6a/200/?on-error=disconnect`). 
So, only payloads that are a number of bytes less in size than the `DEFAULT_MAX_PAYLOAD_SIZE` are currently accepted. 

How about we account for these extra bytes needed for the topic name by adding 150 bytes to the max message size?
`mqttServerOptions.setMaxMessageSize(getConfig().getMaxPayloadSize() + 150);`

Comment (sophokles73) [reply tests/src/test/java/org/eclipse/hono/tests/mqtt/MqttPublishTestBase.java:409]: sounds good to me

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**